iree/turbine/kernel/wave/scheduling/loop_reconstruction.py

```diff
@@ -283,9 +283,11 @@ def push_rotating_registers(
                     f"Mapped orig: {node_map[node]} / mapped: {iter_arg} to stage {mapped_stage}."
                 )
             else:
+                mapped_stage = stage + len(registers) - i - 1
```

Comment:
are the changes in this file related to adding of new mfma intrinscis?

Reply:
So that one is not. There are basically 2 types of changes in this PR:
1. Adding new intrinsics
2. Fixes to scheduling to get scheduling to work with the new intrinsics

Reply:
I understand how adding casts need changes in the graph_util and resources, but not completely sure how the new intrinsics would incite these changes in loop_reconstruction. Can you give a little more context here?

Reply:
So adding the new intrinsic helped uncover a bug in the scheduling code. The issue was that when we push rotating registers into the argument context, the rotating registers need to be pushed into a different stage when emitting the loop body vs the epilogue (prologue doesn't matter because we don't have rotating registers). As a result of this not all the results from the loop body were being used in the epilogue but with the fix, seems to be working now.

Reply:
ah nice! :)